crates/cheatcodes/src/inspector.rs

```diff
@@ -1557,6 +1588,32 @@ impl Cheatcodes {
         }
     }
 
+
+    /// Records debug trace data
+    #[cold]
+    fn record_debug_trace(&mut self, interpreter: &mut Interpreter, recorded_debug_steps: &mut Vec<DebugStep>) {
+        // Record opcodes if `startDebugTraceRecording` has been called
+        let current_opcode = interpreter.current_opcode();
+        let instruction_result = interpreter.instruction_result as u8;
+        let stack_inputs = try_or_continue!(
+            get_stack_inputs_for_opcode(
+                current_opcode, interpreter.stack(),
+            )
+        );
+        let mem_inputs = get_memory_input_for_opcode(
```

Comment:
This is not the right path, we already have a tracer and we should reuse that through CheatcodeExecutor

this can likely just be:
- start: enables tracer at opcode tracing level if not already enabled, saves the current trace index 
- stop: takes the range of traces from saved start to current and abi encodes it 

cc @klkvr

Reply:
yep, I think this could be possible by extending `CheatcodesExecutor` with something like this:
```
fn start_steps_recording(&mut self, ccx: &mut Cheatcodes);
fn get_recorded_step(&mut self, ccx: &mut Cheatcodes) -> Vec<CallTraceStep>;
```

where `CallTraceStep` is coming from revm-inspectors

Reply:
oooh, so if I get this comment right, the idea here is to use `self.tracer` instead where it already have all the tracing needed. And it seems like by `self.tracer.traces().into_nodes()` I can get the `Vec<CallTraceNode>`, where it has the `CallTrace` that I need.

Am i getting this right?

also, I can assume the nodes are append only, so I can just rely on the `idx` field of the `CallTraceNode` to determine the range according to your description above, right?

Reply:
would like to consult a bit, got one problem related to this approach after trying it:

For some background on my local env, I have a repository with forge tests that uses this new cheatcodes and I build the forge locally to use on the test repository. When I try the tracer approach, It panics with the error message: `more traces were filled than started`. I can, however, temporary fix this by turn on the tracing from start in the following function in `crates/evm/evm/src/inspectors/stack.rs` to fix this:
```
pub fn tracing(&mut self, mode: TraceMode) {
        if let Some(config) = mode.into_config() {
            *self.tracer.get_or_insert_with(Default::default).config_mut() = config;
        } else {
            // self.tracer = None;  <-- comment out this line
           // Chage: ensures that the tracing will occur on start.
            self.tracer.get_or_insert_with(Default::default);
        }
    }
 ```
 
 A sample implementation that updates the config is in the same file:
 ```
 impl CheatcodesExecutor for InspectorStackInner {
    fn get_inspector<'a, DB: DatabaseExt>(
        &'a mut self,
        cheats: &'a mut Cheatcodes,
    ) -> impl InspectorExt<DB> + 'a {
        InspectorStackRefMut { cheatcodes: Some(cheats), inner: self }
    }

   // Newly added function here!
    fn start_steps_recording(&mut self, cheats: &mut Cheatcodes) {
        // Ensure the tracer exists and configure it
        let tracer = self.tracer.get_or_insert_with(Default::default);

        tracer.update_config(|_config| TracingInspectorConfig::all());
    }

    .....skip
    
}
```

Wonder if you know any idea/directions to solve the "more traces were filled than started" issue? The current one(turning on the tracer) does not really seems like a great idea....

Reply:
Still having the same issue, I set the trace to use "none()" configuration so make it a dummy one on start instead of having none in the tracer field on start.